### PR TITLE
Fix issues with JTF blocking in codemodel

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2054612.

### CODE REVIEW

File: src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs [hunk 1]
```diff
@@ -39,8 +41,8 @@ private RootCodeModel(CodeModelState state, EnvDTE.Project parent, ProjectId pro
         private Project GetProject()
             => Workspace.CurrentSolution.GetProject(_projectId);
 
-        private Compilation GetCompilation()
-            => GetProject().GetCompilationAsync().Result;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs:43]: so old we're still calling .Result, not even GetAwaiter().GetResult();

File: src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs [hunk 2]
```diff
@@ -127,27 +129,34 @@ EnvDTE.CodeElements ICodeElementContainer<AbstractExternalCodeElement>.GetCollec
...
 
         public EnvDTE.CodeElements CodeElements
-        {
-            get
+            => this.State.ThreadingContext.JoinableTaskFactory.Run(async () =>
             {
-                var compilation = GetCompilation();
+                var compilation = await GetCompilationAsync().ConfigureAwait(true);
+
+                // Need to ensure we're on the UI thread to make the ExternalCodeNamespace.  It creates  UI thread bound
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs:136]:
```suggestion
                // Need to ensure we're on the UI thread to make the ExternalCodeNamespace.  It creates UI thread bound
```

**Review by davkean — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal

File: src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs (hunk 1 above)
Comment (ToddGrun) [reply src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs:43]: Yep, at least 10 years old from the blame.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski For review when you get back.

**Pull request merged**